tests/integration/test_service_manager.py

```diff
@@ -43,6 +46,20 @@ def test___grpc_service_without_discovery_service___start_service___service_host
     _validate_if_service_running_by_making_rpc(port_number)
 
 
+@pytest.mark.parametrize("expect_discovery_service_error_stub", [True])
+def test___grpc_service___start_service_error_registering_measurement___raises_error(
+    grpc_service: GrpcService,
+):
+    with pytest.raises(Exception):
```

Comment:
The fake stub's use of `Exception` hides the fact that your code throws away the exception's derived type.

Please add this to `fake_discovery_service.py`, update the fake stub to use it, and update this test to expect the new exception type. This will demonstrate some of the problems with your proposed changes.

```
class FakeDiscoveryError(Exception):
    pass
```

Reply:
Thanks for this approach! I have added the custom exception `FakeDiscoveryError` and expected it in test.